tests/test_unit/test_workflow.py

```diff
@@ -65,3 +69,36 @@ def test_workflow_from_yaml():
 
     # THEN
     assert workflow == Workflow.from_yaml(workflow.to_yaml())
+
+
+@script()
+def hello():
+    print("hello")
+
+
+def test_workflow_callable_return_is_none():
+    # GIVEN
+    with Workflow(name="w"):
+        returned_from_script = hello()
+
+    # THEN
+    assert returned_from_script is None
+
+
+def test_workflow_callable_container_raises_error():
+    with pytest.raises(InvalidTemplateCall) as e:
```

Comment:
Maybe we should have more negative testing throughout for combining different constructs (Workflows, Scripts, DAGs, Steps)

Reply:
Currently I think only a `ContainerSet` would be another negative example. However I've just realised we should separate out the error for Script/Container as the error says `Callable Template 'whalesay' is not under a Workflow` when it is

Reply:
Well spotted!